### DISCUSSION TITLE
Hide/show some of the points added to the plot as numpy array

### DISCUSSION THREAD

**Discussion by kerim371** [pyvista/pyvista#6673] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
actor = pl.add_points(df[["x", "y", "z"]].values, render_points_as_spheres=True, point_size=10.0)
```

### Description

Adding numpy array as points is done fine.
But the I need to have an access to the points so I could dynamically change the visibility and color of some of them.
I can't understand how can I access these points (spheres) from the given actor and the how do I change the visibility and color of some of them. 

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sun Sep 22 16:17:58 2024 Russia TZ 2 Standard Time

                  OS : Windows (10 10.0.22631 SP0 Multiprocessor Free)
              CPU(s) : 16
             Machine : AMD64
        Architecture : 64bit
         Environment : Python
          GPU Vendor : NVIDIA Corporation
        GPU Renderer : NVIDIA GeForce RTX 3050 Laptop GPU/PCIe/SSE2
         GPU Version : 4.5.0 NVIDIA 546.80
    MathText Support : False

  Python 3.9.11 (tags/v3.9.11:2de452f, Mar 16 2022, 14:33:45) [MSC v.1929 64
  bit (AMD64)]

             pyvista : 0.44.1
                 vtk : 9.3.1
               numpy : 2.0.2
          matplotlib : 3.9.2
              scooby : 0.10.0
               pooch : 1.8.2
              pillow : 10.4.0
             imageio : 2.35.1
             IPython : 8.18.1
            colorcet : 3.1.0
             cmocean : 4.0.3
          ipywidgets : 8.1.5
               scipy : 1.13.1
              meshio : 5.3.5
               trame : 3.6.3
        trame_client : 3.2.5
        trame_server : 3.1.0
           trame_vtk : 2.8.10
       trame_vuetify : 2.7.0
jupyter_server_proxy : 4.4.0
        nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by kerim371** [comment]
I think the easiest way to hide some points is to reacreate the mesh every time hide/show points is needed:
```python
mesh= pv.PolyData(df[["x", "y", "z"]].values)
actor = pl.add_mesh(mesh, point_size=point_size, render_points_as_spheres=True)
```
and to modify point size we can use:
```python
actor.GetProperty().point_size = new_size
```

**Reply by banesullivan** [reply]
I think this is probably the best way to do this (there are "ghost cells" 👻 , but I don't think that will work well here). What I would do is create a copy of the mesh and plot that copy then update the copy in-place with the subset of the data you want to plot:

```py
import numpy as np
import pyvista as pv
import vtk

mesh = pv.PolyData(np.random.rand(100, 3))

plotted = mesh.copy()

pl = pv.Plotter()
pl.add_mesh(plotted, point_size=10, render_points_as_spheres=True)
pl.show()

# ... later on
plotted.copy_from(mesh.remove_cells(list(range(0, 50))))
pl.render()

plotted.copy_from(mesh.remove_cells(list(range(50, 100))))
pl.render()
```

**Reply by kerim371** [reply]
Thank you! Interisting solution
